plugins/jabra-gnp/fu-jabra-gnp-device.c

```diff
@@ -104,24 +118,34 @@ fu_jabra_gnp_device_rx_cb(FuDevice *device, gpointer user_data, GError **error)
 					     error)) {
 		g_prefix_error(error, "failed to read from device: ");
 		return FALSE;
-	} else {
-		if (rx_data->rxbuf[5] == match_buf[5] && rx_data->rxbuf[6] == match_buf[6]) {
-			/* battery report, ignore and rx again */
-			if (!g_usb_device_interrupt_transfer(
-				fu_usb_device_get_dev(FU_USB_DEVICE(self)),
-				0x81,
-				rx_data->rxbuf,
-				FU_JABRA_GNP_BUF_SIZE,
-				NULL,
-				rx_data->timeout,
-				NULL, /* cancellable */
-				error)) {
-				g_prefix_error(error, "failed to read from device: ");
-				return FALSE;
-			}
+	}
+	if (rx_data->rxbuf[2] == match_buf[2] && rx_data->rxbuf[5] == match_buf[5] &&
+	    rx_data->rxbuf[6] == match_buf[6]) {
+		/* battery report, ignore and rx again */
+		if (!g_usb_device_interrupt_transfer(fu_usb_device_get_dev(target),
+						     0x81,
+						     rx_data->rxbuf,
+						     FU_JABRA_GNP_BUF_SIZE,
+						     NULL,
+						     rx_data->timeout,
+						     NULL, /* cancellable */
+						     error)) {
+			g_prefix_error(error, "failed to read from device: ");
+			return FALSE;
 		}
 	}
 
+	if (fu_memcmp_safe((const guint8 *)rx_data->rxbuf,
+			   FU_JABRA_GNP_BUF_SIZE,
+			   0,
```

Comment:
this doesn't fill me with the warm-and-fuzzies; could we do something like:

     typedef struct {
    -       guint8 *rxbuf;
    +       guint8 rxbuf[FU_JABRA_GNP_BUF_SIZE];
            const guint timeout;
     } FuJabraGnpRxData;

...and then do `sizeof(rx_data->rxbuf)` here? then we can also get rid of a few local buffers too. Same comment for FuJabraGnpTxData obviously too.